### PR TITLE
Fix metric `steward_pipelineruns_ongoing_state_duration_periodic_observations_seconds`

### PR DESCRIPTION
**Pull request by anbrsap**
### Description

The duration of an observed pipelinerun's current phase is wrongly calculated based on the start of the pipeline run (`status.startedAt`) instead of the start of the current phase (`status.stateDetails.startedAt`).

Consequently no observations are made for phases `preparing` and `waiting`, because there was no start time yet. For observations of phases `cleaning` and `finished` the duration is too long.

### Submitter checklist

- [x] Change has been tested (on a back-end cluster)
- [x] [changelog.yaml] entry with upgrade notes is prepared and appropriate for the audience affected by the change (users or developer, depending on the change)
- [x] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
    - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) (`version` and `appVersion`)

In case dependencies have been updated:
- N/A Links to external changelogs, since the last release of our component, added to the [changelog.yaml] entry (description).
- N/A Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- N/A Check if dependency updates affect our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There is at least 1 approval for the pull request and no outstanding requests for change
- [x] All voter checks have passed
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] The Pull Request title is understandable and reflects the changes well
- [x] The Pull Request description is understandable and well documented
- [x] [changelog.yaml] entry for this Pull Request has been added
    - [x] Changelog entry contains all required information
    - [x] 'Upgrade notes' are documented in changelog.yaml (if required)

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

File: pkg/runctl/metrics/pipelineruns_periodic_test.go [hunk 1]
```diff
@@ -163,11 +160,11 @@ func Test_pipelineRunsPeriodic_NonNewRun(t *testing.T) {
 
 				run := &stewardapi.PipelineRun{}
 				run.Status.State = state
+				run.Status.StateDetails.State = state
```
Comment (rinckm) [inline pkg/runctl/metrics/pipelineruns_periodic_test.go:163]: Why is this required? 
In [pkg/runctl/metrics/pipelineruns_periodic.go](https://github.com/SAP/stewardci-core/pull/322/files?diff=split&w=0#diff-10097660fa5d0bf4a5c58c34f41882e61de0888c277e5ab8e53fe3309e89b3da) line 76 run.Status.State is used. We should decide to use one of them run.Status.State or run.Status.StateDetails.State

Comment (anbrsap) [reply pkg/runctl/metrics/pipelineruns_periodic_test.go:163]: This information is redundantly contained in a PipelineRunStatus. The metrics implementation may use the one or the other field. Therefore, the test sets both (as in real objects).

Comment (anbrsap) [reply pkg/runctl/metrics/pipelineruns_periodic_test.go:164]: However, it was inconsistent in the two tests. I've added this line in `Test_..._NewRun`, too.

**Review by rinckm — APPROVED** [review]

File: pkg/runctl/metrics/pipelineruns_periodic_test.go [hunk 2]
```diff
@@ -84,14 +84,12 @@ func Test_pipelineRunsPeriodic_NewRun(t *testing.T) {
 
 				run := &stewardapi.PipelineRun{}
 				run.Status.State = tc.state
+				run.Status.StateDetails.State = state
```
Comment (rinckm) [inline pkg/runctl/metrics/pipelineruns_periodic_test.go:87]:
```suggestion
				run.Status.StateDetails.State = tc.state
```

Comment (rinckm) [inline pkg/runctl/metrics/pipelineruns_periodic_test.go:87]:
```suggestion
				run.Status.State = state
				run.Status.StateDetails.State = state
```
Oder auch

Comment (anbrsap) [reply pkg/runctl/metrics/pipelineruns_periodic_test.go]: Done
